### PR TITLE
Fix exception creation during vTable init

### PR DESCRIPTION
**Pull request by gacholio**
Do not create exceptions while holding classTableMutex - instead, pass
the data out to be used after the appropriate locks have been released.

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

**Comment by gacholio** [comment]
Jenkins test sanity

File: runtime/vm/createramclass.cpp [hunk 1]
```diff
@@ -530,8 +540,9 @@ addInterfaceMethods(J9VMThread *vmStruct, J9ClassLoader *classLoader, J9Class *i
 							existing_entry = previous_entry;
 							J9EquivalentEntry * new_entry = (J9EquivalentEntry*) pool_newElement(equivalentSets);
 							if (NULL == new_entry) {
+								/* OOM will be thrown */
 								setNativeOutOfMemoryError(vmStruct, 0, 0);
```
Comment (DanHeidinga) [inline runtime/vm/createramclass.cpp]: `setNativeOutOfMemoryError` should this have been removed like the call above?

Comment (gacholio) [reply runtime/vm/createramclass.cpp]: Arg, yep.

Comment (gacholio) [reply runtime/vm/createramclass.cpp]: Fixed.

File: runtime/vm/createramclass.cpp [hunk 2]
```diff
@@ -559,7 +569,14 @@ addInterfaceMethods(J9VMThread *vmStruct, J9ClassLoader *classLoader, J9Class *i
...
 									J9UTF8 *interfaceClassNameUTF = J9ROMCLASS_CLASSNAME(interfaceClass->romClass);
-									setClassLoadingConstraintOverrideError(vmStruct, J9ROMCLASS_CLASSNAME(romClass), vTableMethodLoader, vTableMethodClassNameUTF, interfaceLoader, interfaceClassNameUTF, interfaceClassNameUTF, J9UTF8_DATA(vTableMethodNameUTF), J9UTF8_LENGTH(vTableMethodNameUTF), J9UTF8_DATA(vTableMethodSigUTF), J9UTF8_LENGTH(vTableMethodSigUTF));
+									/* LinkageError will be thrown */
+									errorData->loader1 = vTableMethodLoader;
+									errorData->class1NameUTF = vTableMethodClassNameUTF;
+									errorData->loader2 = interfaceLoader;
+									errorData->class2NameUTF = interfaceClassNameUTF;
+									errorData->exceptionClassNameUTF = interfaceClassNameUTF;
+									errorData->methodNameUTF = vTableMethodNameUTF;
+									errorData->methodSigUTF = vTableMethodSigUTF;
```
Comment (DanHeidinga) [inline runtime/vm/createramclass.cpp]: for code clarity, should this `goto fail` rather than `goto done`?

**Comment by DanHeidinga** [comment]
Jenkins test sanity

File: runtime/vm/createramclass.cpp (hunk 2 above)
Comment (gacholio) [reply runtime/vm/createramclass.cpp]: Agreed - fixed.

**Pull request merged**